modules/alloy/src/main/ts/ephox/alloy/behaviour/blocking/BlockingApis.ts

```diff
@@ -0,0 +1,80 @@
+import { Optional, Optionals } from '@ephox/katamari';
+import { Attribute } from '@ephox/sugar';
+import * as BehaviourTypes from '../common/BehaviourTypes';
+import * as Behaviour from '../../api/behaviour/Behaviour';
+import { Keying } from '../../api/behaviour/Keying';
+import { Focusing } from '../../api/behaviour/Focusing';
+import { Replacing } from '../../api/behaviour/Replacing';
+import { AlloySpec } from '../../api/component/SpecTypes';
+import { AlloyComponent } from '../../api/component/ComponentApi';
+import * as GuiFactory from '../../api/component/GuiFactory';
+import { BlockingConfig } from './BlockingSchema';
+import { BlockingState } from './BlockingState';
+
+export interface BlockingApis extends BehaviourTypes.BehaviourApisRecord<BlockingConfig, BlockingState> {
+  block: typeof block;
+  unblock: typeof unblock;
+}
+
+// Mark this component as busy, or blocked.
+const block = (
+  component: AlloyComponent,
+  config: BlockingConfig,
+  state: BlockingState,
+  // This works in conjunction with the 'getRoot' function in the config. To
+  // attach a blocker component to the dom, ensure that 'getRoot' returns a
+  // component, and this function returns the specification of the component to
+  // attach.
+  getBusySpec?: (behaviour: Behaviour.AlloyBehaviourRecord) => AlloySpec
+) => {
+  if (!state.getBlocked()) {
+    config.onBlock(component);
+  }
+
+  state.setBlocked(true);
+
+  Attribute.set(component.element, 'aria-busy', true);
```

Comment:
So is the fact that you set this before identifying if there is a valid busySpec and root intentional? Do you want aria-busy to be true, without actually setting any busy elements? Same with block state.

Reply:
Reading the code, it seems like it might be intentional, but just verifying.

Reply:
As per other comments, I wrote this under the assumption that not having busySpec / a block root was a valid configuration option, not a failure mode. If we don't want to support that configuration, I'll change the types so that that state is unreachable

Reply:
I'd be in favour of. making it compulsory... it would be very strange to have something marked as `aria-busy` and not have something visually showing it's blocked/busy.  `getRoot` could still be optional though, with it defaulting to the current component.

Reply:
Worth noting that `Invalidating` that you mentioned you based this off does fallback to the current component if getRoot is `none`:
```
const elem = invalidConfig.getRoot(component).getOr(component.element);
```

- [ ] Make getRoot fallback to the current component, rather than just do nothing.

Reply:
done